### ISSUE TITLE
Array function expressions are unnecessarily parenthesized in parameters

### ISSUE THREAD

**Issue by dead-claudia** [terser/terser#1348]
**Bug report or Feature request?** Feature request

**Version (complete output of `terser -V` or specific git commit)** 5.16.3

**Complete CLI command or `minify()` options used**

```
terser --compress
```

**`terser` input**

```js
foo(() => bar(a, c, e))
```

**`terser` output or error**

```js
foo((()=>bar(a,c,e)));
```

**Expected result**

```js
foo(()=>bar(a,c,e));
```

Caught this while writing up #1347

**Comment by dead-claudia** [comment]
Good point. Still shouldn't be the default since [V8 says people shouldn't be doing that normally](https://v8.dev/blog/preparser#pife).

**Comment by fabiosantoscode** [comment]
Yeah I think it was a mistake to add those parens. IIRC there was some minor performance boost in firefox but I don't remember how good the benchmarks were.
